Source/Util/MapScanner.c

```diff
@@ -0,0 +1,159 @@
+// Copyright CircumScriptor and DarkNeutrino 2021
+#include "MapScanner.h"
+
+#include <Util/Alloc.h>
+#include <Util/Log.h>
+#include <Util/Utlist.h>
+#include <dirent.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+
+// Helper structure for sorting
+typedef struct map_entry {
+    char* name;
+    struct map_entry *next, *prev;
+} map_entry_t;
+
+static int
+is_directory(const char* path)
+{
+    struct stat st;
+    return (stat(path, &st) == 0 && S_ISDIR(st.st_mode));
+}
+
+static int
+file_exists(const char* path)
```

Comment:
Im not sure this is the right place for it. Could it not be in Utils ? The function seems like an function that is a basic support thing for other stuff.

Reply:
Of course !